rotkehlchen/exchanges/coinbase.py

```diff
@@ -160,20 +167,108 @@ def __init__(
             secret=secret,
             database=database,
         )
+
+        self.is_legacy_api_key = self.is_legacy_key(api_key)
         self.apiversion = 'v2'
         self.base_uri = 'https://api.coinbase.com'
         self.msg_aggregator = msg_aggregator
-        self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
-        self.session.headers.update({'CB-VERSION': CB_VERSION})
+        self.host = 'api.coinbase.com'
+
+        if self.is_legacy_api_key:
+            self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
+            self.session.headers.update({'CB-VERSION': CB_VERSION})
+
+    def is_legacy_key(self, api_key) -> bool:
+        """
+        Determines if the provided API key is a legacy one
+        """
+        if LEGACY_RE.match(api_key):
+            log.debug('Legacy Key format!')
+            return True
+        elif NEW_RE.match(api_key):
+            log.debug('New Key format!')
+            return False
+        else:
+            raise AuthenticationError(f'Invalid API key format: {api_key}')
 
     def first_connection(self) -> None:
         self.first_connection_made = True
 
-    def edit_exchange_credentials(self, credentials: ExchangeAuthCredentials) -> bool:
-        changed = super().edit_exchange_credentials(credentials)
-        if credentials.api_key is not None:
-            self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
-        return changed
+    # CDP Method
```

Comment:
As I already mentioned here: https://github.com/rotki/rotki/pull/7903#discussion_r1593533299 removethis comment. You can include it in the docstring itself

```suggestion
```

Reply:
This should be done!